### PR TITLE
update pull-local#16756

### PR DESCRIPTION
**Pull request by heartlock**
Signed-off-by: heartlock 21521209@zju.edu.cn

### CODE REVIEW

**Comment by heartlock** [comment]
ping @vdemeester

File: integration-cli/docker_cli_pull_local_test.go [hunk 1]
```diff
@@ -54,9 +52,7 @@ func (s *DockerRegistrySuite) TestConcurrentPullWholeRepo(c *check.C) {
 		    ENV BAR bar
 		    CMD echo %s
 		`, repo), true)
-		if err != nil {
-			c.Fatal(err)
-		}
+		c.Assert(err, check.IsNil, check.Commentf("%v", err))
```
Comment (cpuguy83) [inline integration-cli/docker_cli_pull_local_test.go]: No need for the Commentf here since the error output will include the value of `err` already. But if there is something being output by this command, then that'd be really awesome here.

File: integration-cli/docker_cli_pull_local_test.go [hunk 2]
```diff
@@ -87,9 +83,7 @@ func (s *DockerRegistrySuite) TestConcurrentPullWholeRepo(c *check.C) {
 	for _, repo := range repos {
 		dockerCmd(c, "inspect", repo)
 		out, _ := dockerCmd(c, "run", "--rm", repo)
-		if strings.TrimSpace(out) != "/bin/sh -c echo "+repo {
-			c.Fatalf("CMD did not contain /bin/sh -c echo %s: %s", repo, out)
-		}
+		c.Assert(strings.TrimSpace(out) != "/bin/sh -c echo "+repo, check.Equals, false, check.Commentf("CMD did not contain /bin/sh -c echo %s: %s", repo, out))
```
Comment (cpuguy83) [inline integration-cli/docker_cli_pull_local_test.go]: Either:
```
c.Assert(strings.TrimSpace(out), checker.Equals, expected)
```
or
```
c.Assert(out, checker.Contains, expected)
```

The current logic is hard to follow, and won't give any useful information in a failure case, since it'll show `true != false` or something instead of `<value> != <expected value>`

File: integration-cli/docker_cli_pull_local_test.go [hunk 3]
```diff
@@ -165,8 +155,6 @@ func (s *DockerRegistrySuite) TestConcurrentPullMultipleTags(c *check.C) {
 	for _, repo := range repos {
 		dockerCmd(c, "inspect", repo)
 		out, _ := dockerCmd(c, "run", "--rm", repo)
-		if strings.TrimSpace(out) != "/bin/sh -c echo "+repo {
-			c.Fatalf("CMD did not contain /bin/sh -c echo %s: %s", repo, out)
-		}
+		c.Assert(strings.TrimSpace(out) != "/bin/sh -c echo "+repo, check.Equals, false, check.Commentf("CMD did not contain /bin/sh -c echo %s; %s", repo, out))
```
Comment (cpuguy83) [inline integration-cli/docker_cli_pull_local_test.go]: Same as the one above like this.

File: integration-cli/docker_cli_pull_local_test.go [hunk 4]
```diff
@@ -2,10 +2,9 @@ package main
 
 import (
 	"fmt"
+	"github.com/go-check/check"
```
Comment (vdemeester) [inline integration-cli/docker_cli_pull_local_test.go]: This import was in the right place :stuck_out_tongue:. We tend to separate the built-in import to the other ones :wink:.

**Comment by vdemeester** [comment]
One nit about imports. Other than that LGTM :wink:

**Comment by heartlock** [comment]
updated @vdemeester

**Comment by heartlock** [comment]
ping @vdemeester

**Comment by runcom** [comment]
LGTM

**Comment by vdemeester** [comment]
:tada:
